src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -2416,6 +2416,67 @@ internal void ILGenForCreateInstance(CodeGenerator ilg, Type type, Type cast, bo
                 }
             }
 
+            Label labelReturn = ilg.DefineLabel();
+            Label labelEndIf = ilg.DefineLabel();
+
+            // Type typeInfo = type.GetTypeInfo();
+            LocalBuilder typeInfo = ilg.DeclareLocal(typeof(TypeInfo), "typeInfo");
+            ilg.Ldc(type);
+            MethodInfo getTypeInfoMehod = typeof(IntrospectionExtensions).GetMethod(
+                  "GetTypeInfo",
+                  CodeGenerator.StaticBindingFlags,
+                  new[] { typeof(Type) }
+                  );
+            ilg.Call(getTypeInfoMehod);
+
+            // IEnumerator e = typeInfo.DeclaredConstructors.GetEnumerator();
+            LocalBuilder enumerator = ilg.DeclareLocal(typeof(IEnumerator), "e");
```

Comment:
I do not know if this is an issue. Nonetheless I've updated to IEnumerator<ConstructorInfo>.